### PR TITLE
Gateway: Add a option to enable keepalive-timeout

### PR DESCRIPTION
**Pull request by eloycoto**
Nginx parameter keepalive-timeout cannot be set at all in the gateway.
Due to is not be able to set as a policy, a new ENV variable is now
enabled to define the timeout.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Added http_proxy policy to use an HTTP proxy in api_backed calls. [THREESCALE-2696](https://issues.jboss.org/browse/THREESCALE-2696) [PR #1080](https://github.com/3scale/APIcast/pull/1080)
 - Option to load service configurations one by one lazily [PR #1099](https://github.com/3scale/APIcast/pull/1099)
 - New maintenance mode policy, useful for maintenance periods. [PR #1105](https://github.com/3scale/APIcast/pull/1105), [THREESCALE-3189](https://issues.jboss.org/browse/THREESCALE-3189)
+- Add a option to enable keepalive_timeout on gateway [THREESCALE-2886](https://issues.jboss.org/browse/THREESCALE-2886) [PR #1106](https://github.com/3scale/APIcast/pull/1106)
```
Comment (dfennessy) [inline CHANGELOG.md]:
```suggestion
- Add an option to enable keepalive_timeout on gateway [THREESCALE-2886](https://issues.jboss.org/browse/THREESCALE-2886) [PR #1106](https://github.com/3scale/APIcast/pull/1106)
```

File: gateway/conf.d/apicast.conf [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# {% if http_keepalive_timeout != empty %}
```
Comment (davidor) [inline gateway/conf.d/apicast.conf]: I thought that the original issue was referring to https://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive_timeout which is configured inside the `upstream` context.

File: gateway/conf.d/apicast.conf [hunk 3]
```diff
@@ -152,8 +152,14 @@ location / {
...
 
-  proxy_ignore_client_abort on;
+  # {% if http_keepalive_timeout != empty %}
+  #   {% capture keepalive_timeout %}
+  #{#}   keepalive_timeout {{ http_keepalive_timeout}};
+  #   {% endcapture %}
+  #   {{ keepalive_timeout | replace: "#{#}", "" }}
+  # {% endif %}
 
+  proxy_ignore_client_abort on;
```
[the remark this answers is not in the document]
Comment (eloycoto) [reply gateway/conf.d/apicast.conf]: I did not add that, it was the diff from insert something over that the diff is different.

[the remark this answers is not in the document]
Comment (davidor) [reply gateway/conf.d/apicast.conf]: Right :sweat_smile:

**Pull request merged**